### DISCUSSION TITLE
How use copilot in proprietary programming language.

### DISCUSSION THREAD

**Discussion by brodao2** [community/community#65937] [Copilot Conversations]
### Select Topic Area

Question

### Body

Hey guys.

I work with a proprietary programming language and would like to leverage that into CoPilot. 
It's possible? How to proceed?

This language is used by more than 15K developers e 30K companies in Brazil and other countries.

Thanks in advance.

**Comment by brodao2** [comment]
I found

GitHub Copilot works on any language, including Java, PHP, Python, JavaScript, Ruby, Go, C#, or C++. **Because it’s been trained on languages in public repositories**, it works for most popular languages, libraries and frameworks. It’s also integrated into your editor, and fast enough to use as you type.

I did simple tests in my proprietary language and it gave some results.

**Reply by Delamater** [reply]
This question is a good one. I work on a global software package with a proprietary development language that we maintain ourselves. The problem is, github copilot makes absolutely wrong coding suggestions for our language. For example, I asked it to optimize a for loop that reads against the database and it suggested I write to a file instead. It has no idea what the syntaxes are and wasn't aware that I was reading a database so it just assumed I was reading a file. 

At this point in the life of github CoPilot, it absolutely doesn't make good choices for unknown languages.
